source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -324,6 +327,8 @@ FilterStatus Router::messageEnd() {
 void Router::onUpstreamData(Buffer::Instance& data, bool end_stream) {
   ASSERT(!upstream_request_->response_complete_);
 
+  recordClusterScopeHistogram({upstream_rs_size_}, Stats::Histogram::Unit::Bytes, data.length());
```

Comment:
I think this is measuring the size of a buffer which may only contain a partial response. To measure the response size we'd have to accumulate data.length() in a member field and then report the histogram and rest the accumulator based on the results of `callbacks_->upstreamData` and `upstream_request_->upgrade_response_->onData` below. Or some variation of that.

Reply:
Ah yes indeed, that'd be bad accounting. Somewhere when writing this I thought registering partial writes would be fine, since we care about them individually. But not really, we care about the response as whole. Will update. Thanks.